gr-uhd/lib/usrp_sink_impl.cc

```diff
@@ -557,6 +562,43 @@ namespace gr {
     } // end tag_work()
 
     void
+    usrp_sink_impl::_async_loop()
+    {
+      ::uhd::async_metadata_t async_md;
+      pmt::pmt_t dic = pmt::make_dict();
+      // poll the async message queue at a liberal interval – no use wasting CPU
+      while(true)
+      {
+        //have a way to stop this thread
+        boost::this_thread::interruption_point();
+        //receive asynchronous messages
+        if (_tx_stream->recv_async_msg(async_md, 0.25)) {
```

Comment:
This call into uhd fails to compile on Ubuntu 14.04 LTS with the apt-get installed UHD 3.005.005.  I'm not sure if you support that combination, or if this is something that needs ABI conditionalization.